tests/Silverback.Integration.Tests.E2E/Mqtt/EncryptionTests.cs

```diff
@@ -70,14 +70,133 @@ public async Task Encryption_SimpleMessages_EncryptedAndDecrypted()
             Helper.Spy.OutboundEnvelopes.Should().HaveCount(2);
             Helper.Spy.OutboundEnvelopes[0].RawMessage.Should().BeOfType<SymmetricEncryptStream>();
             Helper.Spy.OutboundEnvelopes[1].RawMessage.Should().BeOfType<SymmetricEncryptStream>();
+            Helper.Spy.OutboundEnvelopes.ForEach(
+                envelope =>
+                    envelope.Headers.GetValue(DefaultMessageHeaders.EncryptionKeyId).Should().BeNull());
 
             Helper.Spy.InboundEnvelopes.Should().HaveCount(2);
             Helper.Spy.InboundEnvelopes[0].Message.Should().BeEquivalentTo(message1);
             Helper.Spy.InboundEnvelopes[1].Message.Should().BeEquivalentTo(message2);
+            Helper.Spy.InboundEnvelopes.ForEach(
+                envelope =>
+                    envelope.Headers.GetValue(DefaultMessageHeaders.EncryptionKeyId).Should().BeNull());
 
             Helper.GetClientSession("e2e-test").PendingMessagesCount.Should().Be(0);
         }
 
+        [Fact]
+        public async Task Encryption_SimpleMessagesWithKeyRotation_EncryptedAndDecrypted()
+        {
+            var message1 = new TestEventOne { Content = "Message 1" };
+            var message2 = new TestEventTwo { Content = "Message 2" };
+            const string expectedKeyIdentifierForMessage1 = "my-encryption-key-1";
+            const string expectedKeyIdentifierForMessage2 = "my-encryption-key-2";
+
+            Host.ConfigureServices(
+                    services => services
+                        .AddLogging()
+                        .AddSilverback()
+                        .UseModel()
+                        .WithConnectionToMessageBroker(options => options.AddMockedMqtt())
+                        .AddMqttEndpoints(
+                            endpoints => endpoints
+                                .Configure(
+                                    config => config
+                                        .WithClientId("e2e-test")
+                                        .ConnectViaTcp("e2e-mqtt-broker"))
+                                .AddOutbound<TestEventOne>(
+                                    endpoint => endpoint
+                                        .ProduceTo(DefaultTopicName)
+                                        .EncryptUsingAes(AesEncryptionKey, expectedKeyIdentifierForMessage1))
```

Comment:
You are always using the same key though. Wouldn't it be better to use different ones for a realistic test?